### PR TITLE
feat(rome_js_analyze): `noSetterReturn`

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This implements Deno Lint [noSetterReturn](https://lint.deno.land/?q=&all=on#no-setter-return).

## Test Plan

Unit tests included.

### CODE REVIEW

**Review by MichaReiser — APPROVED** [review]

File: crates/rome_js_analyze/src/analyzers/nursery/no_setter_return.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    /// ```
+    pub(crate) NoSetterReturn {
+        version: "11.0.0",
+        name: "noSetterReturn",
+        recommended: false,
+    }
+}
+
+impl Rule for NoSetterReturn {
+    type Query = Ast<JsSetterMember>;
```
[the remark this answers is not in the document]
Comment (Conaclos) [reply crates/rome_js_analyze/src/analyzers/nursery/no_setter_return.rs:79]: Have you examples? If I understand well I can access ancestors via `node.syntax().ancestors()` and it should exists some method to attempt a conversion?

[the remark this answers is not in the document]
Comment (Conaclos) [reply crates/rome_js_analyze/src/analyzers/nursery/no_setter_return.rs]: Thanks for the suggestion. The code is simpler in that way:)

File: crates/rome_js_analyze/src/control_flow/visitor.rs [hunk 2]
```diff
@@ -156,7 +156,7 @@ pub(super) struct FunctionVisitor {
 }
 
 declare_node_union! {
-    JsAnyControlFlowRoot = JsModule
+    pub JsAnyControlFlowRoot = JsModule
```
Comment (leops) [inline crates/rome_js_analyze/src/control_flow/visitor.rs:159]: Nit: this could be `pub(crate)` rather than `pub`

**Review by leops — APPROVED** [review]

**Review by ematipico — CHANGES_REQUESTED** [review]

**Review by ematipico — APPROVED** [review]

**Comment by Conaclos** [comment]
@ematipico I rebased this branch on [noConstructorReturn](https://github.com/rome/tools/pull/3805). This should allow merging `noSetterReturn` this feature just after [noConstructorReturn](https://github.com/rome/tools/pull/3805).

**Pull request merged**